tools/scripts/deploy.sh

```diff
@@ -0,0 +1,132 @@
+#!/bin/sh
+###############################################################
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+# Copyright (c) 2019 Tomer Eliyahu (Intel)
+# This code is subject to the terms of the BSD+Patent license.
+# See LICENSE file for more details.
+###############################################################
+
+scriptdir="$(cd "${0%/*}" || exit; pwd)"
+rootdir="${scriptdir%/*/*}"
+
+dbg() {
+    [ "$VERBOSE" = "true" ] && echo "$@"
+}
+
+err() {
+    printf "\033[1;31m%s\n\033[0m" "$@"
+}
+
+success() {
+    printf "\033[1;32m%s\n\033[0m" "$@"
+}
+
+run() {
+    dbg "$*"
+    "$@" || exit $?
+}
+
+
+usage() {
+    echo "usage: $(basename "$0") [-hvsmdCD]"
+}
+
+main() {
+
+    if ! OPTS=$(getopt -o 'hvdb:i:' --long verbose,help,deploy,build-dir:,target-ip:,target-user:,target-pass:,proxy-ip:,proxy-user:,proxy-pass: -n 'parse-options' -- "$@") ; 
+    then err "Failed parsing options." >&2 ; usage; exit 1 ; 
+    fi
+
+    eval set -- "$OPTS"
+
+    while true; do
+        case "$1" in
+            -v | --verbose)       VERBOSE=true; shift ;;
+            -h | --help)          usage; exit 0; shift ;;
+            -d | --deploy)        DEPLOY=true; shift ;;
+            -b | --build-dir)     BUILD_DIR="$2"; shift; shift ;;
+            --proxy-ip)           PROXY_IP="$2"; shift; shift ;;
+            --proxy-user)         PROXY_USER="$2"; shift; shift ;;
+            --proxy-pass)         PROXY_PASS="$2"; shift; shift ;;
+            --target-ip)          TARGET_IP="$2"; shift; shift ;;
+            --target-user)        TARGET_USER="$2"; shift; shift ;;
+            --target-pass)        TARGET_PASS="$2"; shift; shift ;;
+            -- ) shift; break ;;
+            * ) err "unsupported argument $1"; usage; exit 1 ;;
+        esac
+    done
+
+    echo "Start"
+    tmpdir=$(mktemp -d)
+    mkdir -p "${tmpdir}"/opt/prplmesh
+    mkdir -p "${tmpdir}"/usr
+
+    cp -r "${BUILD_DIR}"/install/bin "${tmpdir}"/opt/prplmesh
+    cp -r "${BUILD_DIR}"/install/config "${tmpdir}"/opt/prplmesh
+    cp -r "${BUILD_DIR}"/install/lib "${tmpdir}"/usr/
+    cp -r "${BUILD_DIR}"/install/scripts "${tmpdir}"/opt/prplmesh
+    cp -r "${BUILD_DIR}"/install/share "${tmpdir}"/opt/prplmesh
+
+    dbg "create prplmesh.tar.gz..."
+    tar czvf "${BUILD_DIR}"/prplmesh.tar.gz -C "${tmpdir}"/ .
+    rm -rf "${tmpdir}"
+
+    [ "$DEPLOY" = "true" ] && {
+
+        echo "local md5sum"
+        run md5sum "${BUILD_DIR}"/prplmesh.tar.gz
+
+        if [ -n "$PROXY_IP" ] && [ -n "$PROXY_USER" ] && [ -n "$PROXY_PASS" ]; then
+
+            echo "scp ${BUILD_DIR}/prplmesh.tar.gz to ${PROXY_USER}@${PROXY_IP}"
+            run sshpass -p "${PROXY_PASS}" scp "${SSH_OPTIONS}" "${BUILD_DIR}"/prplmesh.tar.gz "${PROXY_USER}"@"${PROXY_IP}":/home/"${PROXY_USER}"/
+
```

Comment:
You don't need to copy anything on the proxy. You can use ProxyJump for SSH as well as for scp commands to copy it directly to the target (at least with OpenSSH 7.3+; I have no idea about sshpass but it looks like a wrapper over the ssh commands).
With SSH it means using `-J`, with scp I think you have to use `-oProxyjump=<target>`.

Reply:
That's a great feedback, thank you! Changed to use ProxyJump.